File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int r=0,a=0,b=1;
    for(int i=2;i<n+1;i++) {
        r=a;
        a=b;
        b=(r+b)%1234567;
    }
    return b;
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        int tmp=0,a=1,b=1;
        for(int i=3;i<n+1;i++) {
            tmp=a;
            a=b;
            b=(tmp+b)%1234567;
        }
        return b;
    }
}